【バグ】 With WORLD_SCORE disabled, compilation fails after #1968

## 環境

- OS: Linux X11 (Ubuntu 20.04.3 LTS as provided by GitHub's ubuntu-latest runner)
- バージョン:  3.0.0Alpha50 after #1968

## バグの詳細

When world score reports are disabled, send_world_score() in src/core/scores.cpp fails to compile because of an undeclared symbol, display_player.

## 再現手順

Configured and built the source code with:

1. ./bootstrap
2. ./configure --disable-worldscore --disable-pch
3. make

That failed with this message from gcc:

core/scores.cpp: In function ‘bool send_world_score(PlayerType*, bool)’:
core/scores.cpp:165:11: error: ‘display_player’ was not declared in this scope
165 |     (void)display_player;
       |           ^~~~~~~~~~~~~~

## 期待される正常な動作

Compilation works regardless of whether WORLD_SCORE is set or not.

## 補足事項

Report from the failed GitHub action can be found here, https://github.com/backwardsEric/hengband/runs/4765458782?check_suite_focus=true .

--- a/src/core/scores.cpp
+++ b/src/core/scores.cpp
@@ -162,7 +162,6 @@ bool send_world_score(PlayerType *current_player_ptr, bool do_send)
 #else
     (void)current_player_ptr;
     (void)do_send;
-    (void)display_player;
 #endif
     return true;
 }

